src/ORM/Search/SearchContext.php

```diff
@@ -93,7 +94,7 @@ class SearchContext
     public function __construct($modelClass, $fields = null, $filters = null)
     {
         $this->modelClass = $modelClass;
-        $this->fields = ($fields) ? $fields : new FieldList();
+        $this->fields = ($fields) ? $fields : null;
```

Comment:
This change and the one in `getFields()` effectively cancel each other out from a public API perspective - but it allows us to check whether `$this->fields` has been set or not in `getSearchFields()`.